hls/decode.go

```diff
@@ -19,7 +19,9 @@ func (p *MasterPlaylist) Parse(reader io.Reader) error {
 
 	// Parsed Data
 	var renditions []*Rendition
-	variant := &Variant{}
+	variant := &Variant{
```

Comment:
When making these changes, I found that the decode methods below actually created new structs, we could make those decode methods part of the struct, maybe implemented as `UnmarshalText(text []byte) error` to implement `encoding.TextUnmarshaller` interface?